code/modules/reagents/reagent_containers/ladle.dm

```diff
@@ -0,0 +1,53 @@
+/obj/item/reagent_containers/ladle
+	name = "ladle"
+	desc = "A serving ladle. Holds 30u."
+	icon = 'icons/obj/kitchen.dmi'
+	icon_state = "ladle"
+	amount_per_transfer_from_this = 10
+	possible_transfer_amounts = list(10, 15, 30)
+	w_class = ITEMSIZE_SMALL
+	volume = 30
+	force = 6
+	flags = OPENCONTAINER | NOBLUDGEON
+	drop_sound = 'sound/items/drop/glass.ogg'
+	pickup_sound = 'sound/items/pickup/glass.ogg'
+	center_of_mass = list("x"=14, "y"=6)
+
+/obj/item/reagent_containers/ladle/afterattack(var/obj/target, var/mob/user, var/flag)
+	if(!target.is_open_container() || !flag)
+		return ..(target, user, flag)
+	if(reagents.total_volume)
+		if(!target.reagents.get_free_space())
+			to_chat(user, SPAN_NOTICE("[target] is full."))
+			return TRUE
+		var/trans = reagents.trans_to(target, amount_per_transfer_from_this) //sprinkling reagents on generic non-mobs
+		user.visible_message(
+			"<b>[user]</b> pours into [target] from [src].",
+			SPAN_NOTICE("You transfer [trans] units of the solution.")
+		)
+		return TRUE
+	else
+		if(!target.reagents || !target.reagents.total_volume)
```

Comment:
If !target.reagents is true, we already runtime'd back at line 20.

Reply:
This one is fine probably. You *could* reshuffle things so we don't check target.reagents twice potentially, but... up to you.